exercises/src/main/scala/fpinscala/iomonad/IO.scala

```diff
@@ -306,18 +308,41 @@ object IO3 {
                                f: A => Free[F, B]) extends Free[F, B]
 
   // Exercise 1: Implement the free monad
-  def freeMonad[F[_]]: Monad[({type f[a] = Free[F,a]})#f] = ???
+  def freeMonad[F[_]]: Monad[({type f[a] = Free[F,a]})#f] = new Monad[({type f[a] = iomonad.Free[F, a]})#f] {
```

Comment:
I agree it's sounds interesting, and I'm sure I would use it on real code. But the type defined like that was already in the book's repo, it's autocompletion, I didn't write that.